src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/DebugProfileDebugTargetGenerator.cs

```diff
@@ -0,0 +1,120 @@
+    [AppliesTo(ProjectCapability.LaunchProfiles)]
+    [Export(typeof(IDynamicDebugTargetsGenerator))]
+    [ExportMetadata("Name", "DebugProfileProvider")]
+    internal class DebugProfileDebugTargetGenerator : ProjectValueDataSourceBase<IReadOnlyList<IEnumValue>>, IDynamicEnumValuesProvider, IDynamicDebugTargetsGenerator
+    {
+        private IReceivableSourceBlock<IProjectVersionedValue<IReadOnlyList<IEnumValue>>> _publicBlock;
+
+        // Represents the link to the launch profiles
+        private IDisposable _launchProfileProviderLink;
+
+        // Represents the link to our source provider
+        private IDisposable _debugProviderLink;
+
+        [ImportingConstructor]
+        public DebugProfileDebugTargetGenerator(UnconfiguredProject unconfiguredProject, ILaunchSettingsProvider launchSettingProvider)
+            : base(unconfiguredProject.Services)
+        {
+            LaunchSettingProvider = launchSettingProvider;
+        }
+
+        private NamedIdentity _dataSourceKey = new NamedIdentity();
+        public override NamedIdentity DataSourceKey
+        {
+            get { return _dataSourceKey; }
+        }
+
+        /// <inheritdoc/>
+        private int _dataSourceVersion;
+        public override IComparable DataSourceVersion
+        {
+            get { return _dataSourceVersion; }
+        }
+
+        /// <inheritdoc/>
+        public override IReceivableSourceBlock<IProjectVersionedValue<IReadOnlyList<IEnumValue>>> SourceBlock
+        {
+            get
+            {
+                this.EnsureInitialized();
+                return _publicBlock;
+            }
+        }
+
+        private ILaunchSettingsProvider LaunchSettingProvider{ get; }
+
+
+        /// <summary>
+        /// This providers access to the class which creates the list of browser values..
+        /// </summary>
+        public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<Microsoft.Build.Framework.XamlTypes.NameValuePair> options)
+        {
+            return Task.FromResult<IDynamicEnumValuesGenerator>(new DebugProfileEnumValuesGenerator(this.LaunchSettingProvider));
+        }
+
+
+        protected override void Initialize()
+        {
+            var debugProfilesBlock = new TransformBlock<ILaunchSettings, IProjectVersionedValue<IReadOnlyList<IEnumValue>>>(
+                update =>
+                {
+                    // Compute the new enum values from the profile provider
+                   var generatedResult = DebugProfileEnumValuesGenerator.GetEnumeratorEnumValues(update).ToImmutableList();
+                    _dataSourceVersion++;
+                    var dataSources = ImmutableDictionary<NamedIdentity, IComparable>.Empty.Add(DataSourceKey, DataSourceVersion);
+                    return new ProjectVersionedValue<IReadOnlyList<IEnumValue>>(generatedResult, dataSources);
+                });
+
+            var broadcastBlock = new BroadcastBlock<IProjectVersionedValue<IReadOnlyList<IEnumValue>>>(b => b);
```

Comment:
I believe the ctor argument can just be null, as we don't need to clone the enum here.